### PR TITLE
feat(types/guild): add computePruneCount to PruneOptions

### PR DESCRIPTION
**Pull request by ayntee**
Reference: https://discord.com/developers/docs/resources/guild#begin-guild-prune

### CODE REVIEW

File: src/api/handlers/guild.ts [hunk 1]
```diff
@@ -558,7 +565,7 @@ export async function pruneMembers(guildID: string, options: PruneOptions) {
 
   const result = await RequestManager.post(
     endpoints.GUILD_PRUNE(guildID),
-    { ...options, include_roles: options.roles.join(",") },
+    { ...options, include_roles: roles.join(",") },
```
Comment (ayntee) [inline src/api/handlers/guild.ts:568]:
```suggestion
    { ...options, include_roles: roles },
```

File: src/types/guild.ts [hunk 2]
```diff
@@ -547,9 +547,11 @@ export interface PrunePayload {
 }
 
 export interface PruneOptions {
-  /** number of days to count prune for (1 - 30). Defaults to 7 days. */
+  /** Number of days to prune (1-30). Default: 7 */
   days: number;
-  /** Include members with these role ids */
+  /** Whether 'pruned' is returned, discouraged for large guilds. Default: true */
+  compute_prune_count: string[];
+  /** role(s) to include */
```
Comment (ayntee) [inline src/types/guild.ts:554]:
```suggestion
  /** Role(s) to include */
```

File: src/api/handlers/guild.ts [hunk 3]
```diff
@@ -546,8 +546,15 @@ export async function getPruneCount(guildID: string, options: PruneOptions) {
   return result.pruned;
 }
 
-/** Begin pruning all members in the given time period */
-export async function pruneMembers(guildID: string, options: PruneOptions) {
+/**
+ * Begin a prune operation. Requires the KICK_MEMBERS permission. Returns an object with one 'pruned' key indicating the number of members that were removed in the prune operation. For large guilds it's recommended to set the compute_prune_count option to false, forcing 'pruned' to null. Fires multiple Guild Member Remove Gateway events.
```
Comment (Skillz4Killz) [inline src/api/handlers/guild.ts:550]: i think this should use camelCase prop names

File: src/types/guild.ts [hunk 4]
```diff
@@ -547,9 +547,11 @@ export interface PrunePayload {
...
 
 export interface PruneOptions {
-  /** number of days to count prune for (1 - 30). Defaults to 7 days. */
+  /** Number of days to prune (1-30). Default: 7 */
   days: number;
-  /** Include members with these role ids */
+  /** Whether 'pruned' is returned, discouraged for large guilds. Default: true */
+  compute_prune_count: string[];
+  /** Role(s) to include */
   roles: string[];
```
Comment (Skillz4Killz) [inline src/types/guild.ts:555]: shouldnt this be optional? im not sure

File: src/types/guild.ts [hunk 5]
```diff
@@ -547,9 +547,11 @@ export interface PrunePayload {
 }
 
 export interface PruneOptions {
-  /** number of days to count prune for (1 - 30). Defaults to 7 days. */
+  /** Number of days to prune (1-30). Default: 7 */
   days: number;
-  /** Include members with these role ids */
+  /** Whether 'pruned' is returned, discouraged for large guilds. Default: true */
```
Comment (Skillz4Killz) [inline src/types/guild.ts:552]: default true?? but is string[] type??? also camel case?

**Review by Skillz4Killz — CHANGES_REQUESTED** [review]

File: src/types/guild.ts (hunk 4 above)
Comment (ayntee) [reply src/types/guild.ts:555]: @Skillz4Killz it is not mentioned in the documentation, I assume it is optional.

File: src/types/guild.ts (hunk 5 above)
Comment (ayntee) [reply src/types/guild.ts:552]: Good catch, I think this is supposed to be a boolean

File: src/types/guild.ts [hunk 6]
```diff
@@ -547,9 +547,11 @@ export interface PrunePayload {
...
+  compute_prune_count: string[];
```
Comment (ayntee) [inline src/types/guild.ts:553]:
```suggestion
  compute_prune_count: boolean;
```

File: src/types/guild.ts [hunk 7]
```diff
@@ -547,9 +547,11 @@ export interface PrunePayload {
 }
 
 export interface PruneOptions {
-  /** number of days to count prune for (1 - 30). Defaults to 7 days. */
+  /** Number of days to prune (1-30). Default: 7 */
   days: number;
-  /** Include members with these role ids */
+  /** Whether 'pruned' is returned, discouraged for large guilds. Default: true */
+  compute_prune_count: boolean;
```
Comment (ayntee) [inline src/types/guild.ts:553]:
```suggestion
  computePruneCount: boolean;
```

File: src/api/handlers/guild.ts [hunk 8]
```diff
@@ -546,8 +546,15 @@ export async function getPruneCount(guildID: string, options: PruneOptions) {
...
+ * 
+ * By default, prune will not remove users with roles. You can optionally include specific roles in your prune by providing the roles (resolved to include_roles internally) parameter. Any inactive user that has a subset of the provided role(s) will be included in the prune and users with additional roles will not.
+ */
+export async function pruneMembers(
+  guildID: string,
+  { roles, ...options }: PruneOptions,
```
Comment (ayntee) [inline src/api/handlers/guild.ts:556]:
```suggestion
  { roles, computePruneCount, ...options }: PruneOptions,
```

File: src/api/handlers/guild.ts [hunk 9]
```diff
@@ -558,7 +565,7 @@ export async function pruneMembers(guildID: string, options: PruneOptions) {
 
   const result = await RequestManager.post(
     endpoints.GUILD_PRUNE(guildID),
-    { ...options, include_roles: options.roles.join(",") },
+    { ...options, include_roles: roles },
```
Comment (ayntee) [inline src/api/handlers/guild.ts:568]:
```suggestion
    { ...options, compute_prune_count: computePruneCount, include_roles: roles },
```

File: src/api/handlers/guild.ts (hunk 3 above)
Comment (ayntee) [inline src/api/handlers/guild.ts:550]:
```suggestion
 * Begin a prune operation. Requires the KICK_MEMBERS permission. Returns an object with one 'pruned' key indicating the number of members that were removed in the prune operation. For large guilds it's recommended to set the computePruneCount option to false, forcing 'pruned' to null. Fires multiple Guild Member Remove Gateway events.
```

File: src/types/guild.ts (hunk 4 above)
Comment (itohatweb) [reply src/types/guild.ts:555]: Actually all options should be optional

File: src/api/handlers/guild.ts [hunk 10]
```diff
@@ -558,7 +565,7 @@ export async function pruneMembers(guildID: string, options: PruneOptions) {
 
   const result = await RequestManager.post(
     endpoints.GUILD_PRUNE(guildID),
-    { ...options, include_roles: options.roles.join(",") },
+    { ...options, compute_prune_count: computePruneCount, include_roles: roles },
```
Comment (itohatweb) [inline src/api/handlers/guild.ts:568]: Maybe add this comment?

```suggestion
    // TODO: v11 use camel to snake function
    { ...options, compute_prune_count: computePruneCount, include_roles: roles },
```

Comment (ayntee) [reply src/api/handlers/guild.ts:568]: Those functions do not really exist...

File: src/types/guild.ts (hunk 4 above)
Comment (ayntee) [reply src/types/guild.ts:555]: Any reference about whether they should be optional or not?

Comment (itohatweb) [reply src/types/guild.ts:555]: So Discord specifies default values in the docs so all should be optional
`include_roles`:
> You can optionally include specific roles in your prune by providing the `include_roles` parameter

Comment (ayntee) [reply src/types/guild.ts:555]: > So Discord specifies default values in the docs so all should be optional
> `include_roles`:
> 
> > You can optionally include specific roles in your prune by providing the `include_roles` parameter

That refers to include_role field, not `compute_prune_count`

File: src/types/guild.ts [hunk 11]
```diff
@@ -547,9 +547,11 @@ export interface PrunePayload {
 }
 
 export interface PruneOptions {
-  /** number of days to count prune for (1 - 30). Defaults to 7 days. */
+  /** Number of days to prune (1-30). Default: 7 */
   days: number;
-  /** Include members with these role ids */
+  /** Whether 'pruned' is returned, discouraged for large guilds. Default: true */
+  computePruneCount: boolean;
```
Comment (ayntee) [inline src/types/guild.ts:553]:
```suggestion
  computePruneCount?: boolean;
```

File: src/types/guild.ts [hunk 12]
```diff
@@ -547,9 +547,11 @@ export interface PrunePayload {
 }
 
 export interface PruneOptions {
-  /** number of days to count prune for (1 - 30). Defaults to 7 days. */
+  /** Number of days to prune (1-30). Default: 7 */
   days: number;
```
Comment (ayntee) [inline src/types/guild.ts:551]:
```suggestion
  days?: number;
```

File: src/types/guild.ts [hunk 13]
```diff
@@ -547,9 +547,11 @@ export interface PrunePayload {
...
 export interface PruneOptions {
-  /** number of days to count prune for (1 - 30). Defaults to 7 days. */
-  days: number;
-  /** Include members with these role ids */
+  /** Number of days to prune (1-30). Default: 7 */
+  days?: number;
+  /** Whether 'pruned' is returned, discouraged for large guilds. Default: true */
+  computePruneCount: boolean;
+  /** Role(s) to include */
   roles: string[];
```
Comment (ayntee) [inline src/types/guild.ts:555]:
```suggestion
  roles?: string[];
```

File: src/api/handlers/guild.ts [hunk 14]
```diff
@@ -546,8 +546,15 @@ export async function getPruneCount(guildID: string, options: PruneOptions) {
...
+/**
+ * Begin a prune operation. Requires the KICK_MEMBERS permission. Returns an object with one 'pruned' key indicating the number of members that were removed in the prune operation. For large guilds it's recommended to set the computePruneCount option to false, forcing 'pruned' to null. Fires multiple Guild Member Remove Gateway events.
+ * 
+ * By default, prune will not remove users with roles. You can optionally include specific roles in your prune by providing the roles (resolved to include_roles internally) parameter. Any inactive user that has a subset of the provided role(s) will be included in the prune and users with additional roles will not.
+ */
+export async function pruneMembers(
+  guildID: string,
+  { roles, computePruneCount, ...options }: PruneOptions,
+) {
   if (options.days < 1) throw new Error(Errors.PRUNE_MIN_DAYS);
```
Comment (ayntee) [inline src/api/handlers/guild.ts:558]:
```suggestion
  if (options.days && options.days < 1) throw new Error(Errors.PRUNE_MIN_DAYS);
```

File: src/api/handlers/guild.ts [hunk 15]
```diff
@@ -546,9 +546,16 @@ export async function getPruneCount(guildID: string, options: PruneOptions) {
...
+ * Begin a prune operation. Requires the KICK_MEMBERS permission. Returns an object with one 'pruned' key indicating the number of members that were removed in the prune operation. For large guilds it's recommended to set the computePruneCount option to false, forcing 'pruned' to null. Fires multiple Guild Member Remove Gateway events.
+ * 
+ * By default, prune will not remove users with roles. You can optionally include specific roles in your prune by providing the roles (resolved to include_roles internally) parameter. Any inactive user that has a subset of the provided role(s) will be included in the prune and users with additional roles will not.
+ */
+export async function pruneMembers(
+  guildID: string,
+  { roles, computePruneCount, ...options }: PruneOptions,
+) {
+  if (options.days && options.days < 1) throw new Error(Errors.PRUNE_MIN_DAYS);
   if (options.days > 30) throw new Error(Errors.PRUNE_MAX_DAYS);
```
Comment (ayntee) [inline src/api/handlers/guild.ts:559]:
```suggestion
  if (options.days && options.days > 30) throw new Error(Errors.PRUNE_MAX_DAYS);
```

File: src/api/handlers/guild.ts (hunk 10 above)
Comment (ayntee) [inline src/api/handlers/guild.ts:568]:
```suggestion
    {
        ...options,
        compute_prune_count: computePruneCount,
        include_roles: roles,
    },
```

**Review by itohatweb — APPROVED** [review]

**Review by Skillz4Killz — APPROVED** [review]

**Review by itohatweb — APPROVED** [review]

File: src/types/guild.ts [hunk 16]
```diff
@@ -547,10 +547,12 @@ export interface PrunePayload {
 }
 
 export interface PruneOptions {
-  /** number of days to count prune for (1 - 30). Defaults to 7 days. */
-  days: number;
-  /** Include members with these role ids */
-  roles: string[];
+  /** Number of days to prune (1-30). Default: 7 */
+  days?: number;
```
Comment (Skillz4Killz) [inline src/types/guild.ts:551]:
```suggestion
  days?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24 | 25 | 26 | 27 | 28 | 29 | 30;
```

**Review by Skillz4Killz — DISMISSED** [review]

**Review by itohatweb — APPROVED** [review]

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**